src/html5/parser.rs

```diff
@@ -1203,7 +1203,7 @@ impl<'chars> Html5Parser<'chars> {
                     Token::EndTag { name, .. }
                         if name == "tbody" || name == "tfoot" || name == "thead" =>
                     {
-                        if !self.is_in_scope(name, Scope::Table) {
+                        if !self.is_in_scope(name, HTML_NAMESPACE, Scope::Table) {
```

Comment:
There was an issue in is_in_scope where we did check for a tag, but DIDN'T check the namespace. In these specific tests, the tag was "TR", but the namespace was NOT html, but mathml.   This is changed so is_in_scope also received the namespace on where to check. This is always HTML though, but it's more flexible to have it this was..  a "tag" by itself says nothing, and now we are more flexible.